### PR TITLE
Update ObservableList's more conservatively.

### PR DESCRIPTION
**Pull request by frenchy64**
TabPane's seem to respond better to this when subscribed
to :tabs.

### CODE REVIEW

File: src/cljfx/mutator.clj [hunk 1]
```diff
@@ -58,8 +58,21 @@
...
 
 (defn observable-list [get-list-fn]
-  (let [set-all! #(.setAll ^ObservableList (get-list-fn %1)
-                           ^Collection %2)]
+  (let [set-all! (fn [instance coll]
+                   (let [l ^ObservableList (get-list-fn instance)
+                         size (.size l)
+                         csize (count coll)]
+                     ;; update the first (count coll) elements in ObservableList
+                     (reduce (fn [_ [idx e]]
```
Comment (vlaaad) [inline src/cljfx/mutator.clj]: We don't really care about accumulator, I think `doseq` makes more sense here

File: src/cljfx/mutator.clj [hunk 2]
```diff
@@ -58,8 +58,21 @@
...
+                               (if (< idx size)
+                                 (when (not (identical? e (.get l idx)))
+                                   (.set l idx e))
+                                 (.add l idx e)))
+                             nil
+                             (map-indexed vector coll))
```
Comment (vlaaad) [inline src/cljfx/mutator.clj]: I'd prefer to not create an intermediate collection, transducers maybe? Or rewrite as `loop` for extra efficiency?

**Comment by frenchy64** [comment]
This might be fixed upstream, I haven't verified yet.

https://bugs.openjdk.java.net/browse/JDK-8222457

**Comment by frenchy64** [comment]
Assuming this has been fixed upstream by now.

**Pull request closed without merging**